src/shared/pages/team/editTeam/editTeam.tsx

```diff
@@ -41,24 +68,53 @@ const EditTeam: React.FC<EditTeamProps> = ({ csrfToken, history, match }) => {
             });
+        getUnitForTeam(teamId)
+            .then(unit => dispatch({ type: 'SetUnitInitial', payload: { label: unit.displayName, value: unit.type } }))
+            .catch(() => {
+                setErrorMessage(new ErrorMessage(LOAD_TEAM_ERROR_DESCRIPTION, GENERAL_ERROR_TITLE));
+            });
     }, []);
 
-    const handleSubmit = (event: React.FormEvent) => {
+    const handleSubmit = async (event: React.FormEvent) => {
         event.preventDefault();
         clearErrors();
 
+        const successMessages: Array<string> = [];
+        let hasErrors = false;
+
         if (validate(validationSchema, state, addFormError)) {
-            updateTeamName(teamId, state.newDisplayName).then(() => {
-                history.push(`/team-view/${teamId}`, { successMessage: `Team name changed from ${state.currentDisplayName} to ${state.newDisplayName}.` });
-            }).catch((error) => {
-                if (error && error.response && error.response.status === 409) {
-                    setErrorMessage(
-                        new ErrorMessage(TEAM_RENAME_FAILED_NAME_ALREADY_EXISTS, GENERAL_ERROR_TITLE)
-                    );
-                } else {
-                    setErrorMessage(new ErrorMessage(TEAM_RENAME_FAILED_UNKNOWN_ERROR, GENERAL_ERROR_TITLE));
-                }
-            });
+            if (state.newDisplayName !== state.currentDisplayName) {
+                await updateTeamName(teamId, state.newDisplayName).then(() => {
+                    successMessages.push(`Team name changed from ${state.currentDisplayName} to ${state.newDisplayName}.`);
+                    dispatch({ type: 'SetCurrentTeamName', payload: state.newDisplayName });
+                }).catch((error) => {
+                    hasErrors = true;
+                    if (error && error.response && error.response.status === 409) {
+                        setErrorMessage(
+                            new ErrorMessage(TEAM_RENAME_FAILED_NAME_ALREADY_EXISTS, GENERAL_ERROR_TITLE)
+                        );
+                    } else {
+                        setErrorMessage(new ErrorMessage(TEAM_RENAME_FAILED_UNKNOWN_ERROR, GENERAL_ERROR_TITLE));
+                    }
+                });
+            }
+
+            if (state.unit?.value !== state.initialUnit?.value) { // only make unit update call if unit was changed
+                await updateTeamUnit(teamId, state.unit?.value as string).then(() => {
+                    successMessages.push(`Team unit changed from ${state.initialUnit?.label} to ${state.unit?.label}.`);
+                    dispatch({ type: 'SetUnitInitial',
+                        payload: { label: state.unit?.label || '', value: state.unit?.value || '' } });
+                }).catch(() => {
+                    hasErrors = true;
+                    setErrorMessage(new ErrorMessage(TEAM_UPDATE_FAILED_UNKNOWN_ERROR, GENERAL_ERROR_TITLE));
+                });
+            }
+
+            if(!hasErrors) {
```

Comment:
Is there a tidier way of checking for errors rather than manually setting a separate 'hasErrors' variable, it has potential to be missed in a future catch potentially.